### ISSUE TITLE
Fail to build the package with cython 3.0.0

### ISSUE THREAD

**Issue by Felixoid** [mark9064/pyamdgpuinfo#10]
Hello, I am trying to build the package on Arch linux as following, and it fails:

```
> python -m build --wheel --no-isolation
* Getting build dependencies for wheel...
Compiling pyamdgpuinfo/_pyamdgpuinfo.pyx because it changed.
[1/1] Cythonizing pyamdgpuinfo/_pyamdgpuinfo.pyx

Error compiling Cython file:
------------------------------------------------------------
...
        self.thread_args.results = <uint32_t*>calloc(buffer_size_in_ticks, sizeof(uint32_t))
        if not self.thread_args.results:
            raise MemoryError()
        self.thread_args.device_handle = &self.device_handle
        self.utilisation_polling = True
        if pthread_cy.pthread_create(&tid, &attr, poll_registers, self.thread_args) != 0:
                                                  ^
------------------------------------------------------------

pyamdgpuinfo/_pyamdgpuinfo.pyx:382:50: Cannot assign type 'void *(void *) except * nogil' to 'void *(void *) noexcept nogil'
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
    return hook(config_settings)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
    return self._get_build_requires(config_settings, requirements=['wheel'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
    self.run_setup()
  File "/usr/lib/python3.11/site-packages/setuptools/build_meta.py", line 488, in run_setup
    self).run_setup(setup_script=setup_script)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
    exec(code, locals())
  File "<string>", line 22, in <module>
  File "/usr/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 1134, in cythonize
    cythonize_one(*args)
  File "/usr/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 1301, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: pyamdgpuinfo/_pyamdgpuinfo.pyx

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel
```

It's version 2.1.4

Reference to the Arch linux package https://aur.archlinux.org/packages/python-pyamdgpuinfo

**Comment by mark9064** [comment]
Fixed now with 2.1.5 :)
You'll have to wait until the packager updates the arch package (you can flag it out of date on the aur page), until then if you need this working now you can create a python virtualenv and install from pip

**Comment by Felixoid** [comment]
Thanks, it's fixed there too!
